flair/data.py

```diff
@@ -1802,13 +1793,13 @@ def get_spans_from_bio(bioes_tags: List[str], bioes_scores=None) -> List[typing.
         # does this prediction start a new span?
         starts_new_span = False
 
-        # begin and single tags start new spans
-        if bioes_tag[:2] in {"B-", "S-"}:
+        if bioes_tag[:2] in {"B-", "S-"} or (
+            in_span and previous_tag[2:] != bioes_tag[2:] and (bioes_tag[:2] == "I-" or previous_tag[2:] == "S-")
+        ):
+            # B- and S- always start new spans
+            # if the predicted class changes, I- starts a new span
+            # if the predicted class changes and S- was previous tag, start a new span
```

Comment:
Debatable if this improves clarity ;) but many of the other changes are really good!

Reply:
Yes this one is debatable, I considered putting a "#noqa: XYZ" there and keeping the condition as is, but on the other hand this version also has its merits (not duplicate the body in the if - so I just did it the way that ruff was also happy with it :D